pkg/processor/sourceprocessor/source_processor_test.go

```diff
@@ -149,6 +167,97 @@ func assertSpansEqual(t *testing.T, t1 pdata.Traces, t2 pdata.Traces) {
+		"_HOSTNAME":     `sumologic-kubernetes-collection-hostname`,
+	}
+	logAttrs := map[string]string{
+		"PRIORITY":               `6`,
+		"_BOOT_ID":               `7878e140730d4ee89fadd300c929a892`,
+		"_MACHINE_ID":            `1e54ca203e554cda9c944fd7f00e94e1`,
+		"MESSAGE":                `time="2021-09-15T17:31:49.523983251+02:00" level=info msg="ignoring event"module=libcontainerd namespace=moby`,
+		"_TRANSPORT":             `stdout`,
+		"SYSLOG_FACILITY":        `3`,
+		"_UID":                   `0`,
+		"_GID":                   `0`,
+		"_CAP_EFFECTIVE":         `3fffffffff`,
+		"_SELINUX_CONTEXT":       `unconfined`,
+		"_SYSTEMD_SLICE":         `system.slice`,
+		"_STREAM_ID":             `f16d7d0400a44f228554869816ab1dfa`,
+		"SYSLOG_IDENTIFIER":      `dockerd`,
+		"_PID":                   `1061`,
+		"_COMM":                  `dockerd`,
+		"_EXE":                   `/usr/bin/dockerd`,
+		"_CMDLINE":               `/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock`,
+		"_SYSTEMD_CGROUP":        `/system.slice/docker.service`,
+		"fluent.tag":             `host.docker.service`,
+		"_SYSTEMD_INVOCATION_ID": `1b689467e52f4fc4aa95d3c36fa1d7fa`,
+	}
+
+	t.Run("works using existing resource attribute", func(t *testing.T) {
+		config := NewFactory().CreateDefaultConfig().(*Config)
+		config.SourceName = "will-it-work-%{_HOSTNAME}"
+		config.SourceHostKey = "_HOSTNAME"
+
+		pLogs := newLogsDataWithLogs(resourceAttrs, logAttrs)
+
+		sp := newSourceProcessor(config)
+		out, err := sp.ProcessLogs(context.Background(), pLogs)
+		require.NoError(t, err)
+
+		out.ResourceLogs().At(0).Resource().Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+			t.Logf("k %s : v %v\n", k, v.StringVal())
+			return true
+		})
+
+		require.Equal(t, out.ResourceLogs().Len(), 1)
+		resAttrs := out.ResourceLogs().At(0).Resource().Attributes()
+
+		{
+			v, ok := resAttrs.Get("_sourceName")
+			require.True(t, ok)
+			assert.Equal(t, "will-it-work-sumologic-kubernetes-collection-hostname", v.StringVal())
+		}
+
+		{
+			v, ok := resAttrs.Get("_sourceHost")
+			require.True(t, ok)
+			assert.Equal(t, "sumologic-kubernetes-collection-hostname", v.StringVal())
+		}
+	})
+
+	t.Run("does not work using record attribute", func(t *testing.T) {
+		config := NewFactory().CreateDefaultConfig().(*Config)
+		config.SourceName = "will-it-work-%{_CMDLINE}"
+		config.SourceHostKey = "_CMDLINE"
+
+		pLogs := newLogsDataWithLogs(resourceAttrs, logAttrs)
+
+		sp := newSourceProcessor(config)
+		out, err := sp.ProcessLogs(context.Background(), pLogs)
+		require.NoError(t, err)
+
+		out.ResourceLogs().At(0).Resource().Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+			t.Logf("k %s : v %v\n", k, v.StringVal())
+			return true
+		})
+
+		require.Equal(t, out.ResourceLogs().Len(), 1)
+		resAttrs := out.ResourceLogs().At(0).Resource().Attributes()
+
+		{
+			_, ok := resAttrs.Get("_sourceName")
+			require.False(t, ok)
+		}
+
```

Comment:
Why it doesn't set at all?
shouldn't it be `will-it-work-`/`will-it-work-undefined`?

Reply:
That might be a different bug :) or maybe just a different behavior in rendering templates

Reply:
The reason for this is: https://github.com/SumoLogic/sumologic-otel-collector/blob/1c3ef97a999fa3344b3a66287847337b4348104f/pkg/processor/sourceprocessor/attribute_filler.go#L118

Let's park this and sort this out in a separate PR/issue.